src/plugins/mer/mersdkmanager.cpp

```diff
@@ -447,19 +447,22 @@ bool MerSdkManager::addKit(const BuildEngine *buildEngine, const QString &buildT
 
     Core::Id id;
 
+    bool sdkProvided = false;
+
     // Incomplete kits are precreated with sdktool to avoid automatic creation
     // of the Desktop kit
     if (Kit* placeholderKit = kit(buildEngine, buildTargetName)) {
         id = placeholderKit->id();
         KitManager::deregisterKit(placeholderKit);
+        sdkProvided = true;
     }
 
     ToolChainManager::registerToolChain(toolchain.get());
     ToolChainManager::registerToolChain(toolchain_c.get());
     QtVersionManager::addVersion(version.get());
 
     auto initializeKit = [&](Kit *kit) {
-        finalizeKitCreation(buildEngine, buildTargetName, kit);
+        finalizeKitCreation(buildEngine, buildTargetName, kit, sdkProvided);
```

Comment:
Maybe make 'kit' the very first argument when touching this.

Reply:
> More correctly "results in unnecessary assertion failures" :)

yes of course :)